src/Sentry.Maui/Internal/ScreenshotAttachment.cs

```diff
@@ -0,0 +1,52 @@
+namespace Sentry.Maui.Internal;
+
+internal class ScreenshotAttachment : Attachment
+{
+    public ScreenshotAttachment()
+        : this(
+            AttachmentType.Default,
+            new ScreenshotAttachmentContent(),
+            "screenshot.jpg",
+            "image/jpeg")
+    {
+    }
+
+    private ScreenshotAttachment(
+        AttachmentType type,
+        IAttachmentContent content,
+        string fileName,
+        string? contentType)
+        : base(type, content, fileName, contentType)
+    {
+    }
+}
+
+internal class ScreenshotAttachmentContent : IAttachmentContent
+{
+    public Stream GetStream()
+    {
+        try
+        {
+#if !WINDOWS
+            if (Screenshot.Default.IsCaptureSupported)
+            {
+                // This actually runs synchronously (returning Task.FromResult) on the following platforms:
+                // Android: https://github.com/dotnet/maui/blob/3c7b65264d2f341a48db32263a271fd8718cfd23/src/Essentials/src/Screenshot/Screenshot.android.cs#L49
+                // iOS: https://github.com/dotnet/maui/blob/3c7b65264d2f341a48db32263a271fd8718cfd23/src/Essentials/src/Screenshot/Screenshot.ios.cs#L49
+                var screen = Screenshot.Default.CaptureAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+                var stream = screen.OpenReadAsync(ScreenshotFormat.Jpeg).ConfigureAwait(false).GetAwaiter().GetResult();
+
+                return stream;
+            }
+#endif
+        }
+        //In some cases screen capture can throw, for example on Android if the activity is marked as secure.
+        catch (Exception)
+        {
+            return MemoryStream.Null;
+        }
+
+        return MemoryStream.Null;
```

Comment:
No need to go through the `try/catch` on Windows if `try` doesn't do anything.

Also, you can take the `SentryOptions` via the constructor so you can log out problems to ease debugging.
Folks will raise tickets: "Why the screenshot isn't getting attached" and we need to have tools to help us debug

Finally, wrt Android Secure Actitivies is there a way to check? If so, you could do something like: 
```
#if ANDROID
if (Android.Activity.IsSecure) { return
```

```suggestion
        }
        //In some cases screen capture can throw, for example on Android if the activity is marked as secure.
        catch (Exception ex)
        {
            _options.LogError(ex, "Failed to capture screenshot");
            return MemoryStream.Null;
        }
#else
        return MemoryStream.Null;
#endif
```

Reply:
Added logging. As for the secure check, the code can throw in some other cases too so it's safer to have try-catch